lib/daru/accessors/array_wrapper.rb

```diff
@@ -66,27 +66,32 @@ def dup
         ArrayWrapper.new @data.dup, @context
       end
 
+      def compact
+        new_index = (@context.index.to_a - @context.missing_positions)
+        new_index.map { |i| @context[i] }
```

Comment:
Shouldn't `#compact` be returning a data array with the missing elements removed?

How about `@data - @context.missing_values`?